Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/distance.h

```diff
@@ -330,197 +845,125 @@ sample_triangle_mesh(const TriangleMesh& tm,
-      }
-    }
-    // sample edges
-    if (smpl_dgs)
-    {
-      if (nb_points_per_edge == 0 && nb_pts_l_u == 0)
-        nb_pts_l_u = 1. / min_edge_length;
-      for(edge_descriptor ed : edges(tm))
-      {
-        std::size_t nb_points = nb_points_per_edge;
-        if (nb_points == 0)
-        {
-          nb_points = (std::max)(
-            static_cast<std::size_t>( std::ceil( std::sqrt( to_double(
-             squared_distance(get(pmap, source(ed, tm)),
-                              get(pmap, target(ed, tm)) )) )*nb_pts_l_u ) ),
-            std::size_t(1));
-        }
-        // now do the sampling of the edge
-        Random_points_on_segment_3<typename Geom_traits::Point_3, Creator>
-          g(get(pmap, source(ed,tm)), get(pmap, target(ed,tm)));
-        out=std::copy_n(g, nb_points, out);
-      }
-    }
-  }
+}
 
-  // random uniform sampling
-  if (use_rs)
-  {
-    // sample faces
-    if(smpl_fcs)
-    {
-      std::size_t nb_points = choose_parameter(get_parameter(np, internal_np::number_of_points_on_faces), 0);
-      Random_points_in_triangle_mesh_3<TriangleMesh, Vpm, Creator> g(tm, pmap);
-      if (nb_points == 0)
-      {
-        if (nb_pts_a_u == 0.)
-          nb_points = num_vertices(tm);
-        else
-          nb_points = static_cast<std::size_t>(
-            std::ceil(g.mesh_area()*nb_pts_a_u) );
-      }
-      out = std::copy_n(g, nb_points, out);
-    }
-    // sample edges
-    if (smpl_dgs)
-    {
-      std::size_t nb_points =
-        choose_parameter(get_parameter(np, internal_np::number_of_points_on_edges), 0);
-      Random_points_on_edge_list_graph_3<TriangleMesh, Vpm, Creator> g(tm, pmap);
-      if (nb_points == 0)
-      {
-        if (nb_pts_l_u == 0)
-          nb_points = num_vertices(tm);
-        else
-          nb_points = static_cast<std::size_t>(
-            std::ceil( g.mesh_length()*nb_pts_a_u) );
-      }
-      out = std::copy_n(g, nb_points, out);
-    }
-  }
+/** \ingroup PMP_distance_grp
+ * generates points taken on `triangles` and outputs them to `out`, the sampling method
+ * is selected using named parameters.
+ * @tparam PointRange  a model of the concept `RandomAccessContainer` whose value type is the point type.
+ * @tparam TriangleRange a model of the concept `RandomAccessContainer`
+ *                      whose value_type is itself a model of the concept `RandomAccessContainer`
+ *                      whose value_type is `std::size_t`.
+ * @tparam OutputIterator a model of `OutputIterator`
+ *  holding point type objects.
+ *
+ * @param points the points of the soup that will be sampled.
+ * @param triangles a vector containing the triangles of the soup that will be sampled.
+ * @param out output iterator to be filled with sampled points
+ * @param np an optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{geom_traits} a model of `PMPDistanceTraits`. \cgalParamEnd
```

Comment:
Need to express that PointRange::value_type == PMPDistanceTraits::Point_3 (please add a static assertion too).